faculty/clients/api.py

```diff
@@ -0,0 +1,635 @@
+    # TODO: what would be the best way to handle these missing fields for Marshmallow
+    deployment_status = EnumField(
+        DeploymentStatus,
+        by_value=True,
+        data_key="deploymentStatus",
+        load_default=None,
+    )
+    prod_instances = fields.Nested(
+        InstanceSchema,
+        many=True,
+        data_key="prodInstances",
+        load_default=None,
+    )
+    prod_keys = fields.Nested(
+        _APIKeySchema,
+        many=True,
+        data_key="prodKeys",
+        load_default=None,
+    )
+    dev_instances = fields.Nested(
+        InstanceSchema,
+        many=True,
+        data_key="devInstances",
+        load_default=None,
+    )
+
+    @post_load
+    def make_api(self, data, **kwargs):
+        return API(**data)
+
+
+class ListAllAPISchema(BaseSchema):
+    apis = fields.Nested(_APISchema(many=True))
+
+    @post_load
+    def make_list_apis(self, data, **kwargs):
+        # Flatten the API response
+        return data["apis"]
+
+
+class APIClient(BaseClient):
+    """Client for the Faculty API deployment.
+
+    Either build this client with a session directly, or use the
+    :func:`faculty.client` helper function:
+
+    >>> client = faculty.client("api")
+
+    Parameters
+    ----------
+    url : str
+        The URL of the server management service.
+    session : faculty.session.Session
+        The session to use to make requests.
+    """
+
+    SERVICE_NAME = "aperture"
+
+    def create(
+        self,
+        project_id,
+        api_definition,
+        subdomain,
+        name="",
+        description="",
+        environment_ids=[],
+        default_server_size=None,
+    ):
+        """Create a new API.
+
+        Parameters
+        ----------
+        project_id : uuid.UUID
+            The project to create the API in.
+        api_definition : CommandDefinition
+            The API's command definition: WSGI(Flask)/Plumber/Script(Custom).
+        subdomain : str
+            The subdomain where the API should run.
+        name : str
+            The API's name.
+        description : str
+            The API's description field.
+        environment_ids : List[uuid.UUID]
+            The environments to apply to the API's instances.
+        default_server_size : ServerType
+            The default server size to set
+        """
+        payload = {
+            "definition": CommandDefinitionSchema().dump(api_definition)
+        }
+        payload["subdomain"] = subdomain
+        payload["name"] = name
+        payload["description"] = description
+
+        payload["environmentIds"] = environment_ids
+
+        # TODO: turn these into schema handling as well
+        if default_server_size is None:
+            default_server_size = {
+                "instanceSizeType": "custom",
+                "instanceSize": {"milliCpus": 1000, "memoryMb": 4096},
+            }
+        payload["defaultServerSize"] = default_server_size
+
+        return self._post_raw(
```

Comment:
No need (and unconventional in this library) to return the Flask Response object:
```suggestion
        self._post_raw(
```